standardize_mutation_data.py

```diff
@@ -785,6 +785,28 @@ def create_maf_record_from_maf(filename, data, center_name, sequence_source):
     resolve_match_norm_seq_alleles(data, maf_data)
     return maf_data
 
+def detect_file_encoding(filename):
+    """
+        Reads the first million bytes of a file 
+        to detect the type of encoding.
+    """
+    with open(filename, "rb") as data_file:
+        encoding = detect(data_file.read(1000000))
+        return encoding["encoding"]
+
+
+def extract_file_data(filename):
+    """
+        Reads in file data with the appropriate encoding
+        type and returns file data as ascii.
+    """
+    encoding_type = detect_file_encoding(filename)
+    with open(filename, encoding = encoding_type) as data_file:
+        filedata = []
+        for line in data_file.readlines():
+            filedata.append(line.encode("ascii", "ignore").decode("ascii"))
```

Comment:
If I understand this properly, any character which cannot be encoded in ascii will be discarded. Maybe that is ok, but this means silently altering the input. Frequent unicode cases we have seen causing problems before are the substitution of horizontal-stroke characters ((U+2010) -> (U+2015)) for ascii hyphen or so-called "smart-quote" character substitutions ((U+2018), (U+2019), (U+201C), (U+201D)) for ascii apostrophe and quotation mark. This fix would be discarding such symbols, perhaps contracting compound references into a single reference (such as TMPRSS2-ERG -> TMPRSS2ERG, or miR-221-5p -> miR2215p)

When a file has been purged of non-ascii characters, it would probably be a good idea to at least report which files had non-ascii characters which needed to be discarded. A more sophisticated report would show the before / after of each line on which non-ascii characters had been purged (or even the individual field). I think I recommend that we add a list of "detected non-ascii encoded files" to the tool's output message. Then the user could scan the file for non-ascii characters and decide whether any of the lost characters are important enough to justify a manual editing of the input file into ascii format (reverting smart quotes and such) before a re-run.

Reply:
Oh interesting... I didn't realize this was removing those rows... 

The site that has this issue actually has a hard time converting this.  Their VCFs come down the pipe and usually its just one row in a VCF file...I was hoping this code would convert the non-ascii character.

Reply:
I believe the "ignore" option in the encode() call will discard all *characters* which are not encodable in ascii. We could probably alternatively detect and discard rows containing non-ascii encodable characters by returning to a strict encode() mode and catching the exceptions that are thrown on a per-line basis ... discarding the entire line. (if that would be preferable)
We might also be able to provide both modes, and take a command-line argument to decide which policy to use for non-ascii encodable characters.

Reply:
Ahhh, sorry. I misread your comment... I _think_ that it is ok that these characters are just ignored, but I am unsure.

Reply:
added a to-do to report all errors/warnings/logs in a summary report at the end of the run on a file-by-file basis. For now I've added a simple warning message that will report whenever the `detect_file_encoding()` function encounters a file with non-ascii encoding